### PR TITLE
Fix Mismatch Structs on STT REST

### PR DESCRIPTION
**Pull request by dvonthenen**
## Proposed changes

Addresses: https://github.com/deepgram/deepgram-python-sdk/issues/466

Verified working modifying an example:
`examples/speech-to-text/rest/url`

Options:
```
        options: PrerecordedOptions = PrerecordedOptions(
            model="nova-2",
            smart_format=True,
            paragraphs=True,
        )
```

Result:
```
response: {
    "metadata": {
        "transaction_key": "deprecated",
        "request_id": "3c6973b5-d755-4687-97db-938530c82bd6",
        "sha256": "5324da68ede209a16ac69a38e8cd29cee4d754434a041166cda3a1f5e0b24566",
        "created": "2024-09-26T01:37:48.494Z",
        "duration": 17.566313,
        "channels": 1,
        "models": [
            "30089e05-99d1-4376-b32e-c263170674af"
        ],
        "model_info": {
            "30089e05-99d1-4376-b32e-c263170674af": {
                "name": "2-general-nova",
                "version": "2024-01-09.29447",
                "arch": "nova-2"
            }
        }
    },
    "results": {
        "channels": [
            {
                "alternatives": [
                    {
                        "transcript": "Yep. I said it before, and I'll say it again. Life moves pretty fast. You don't stop and look around once in a while, you could miss it.",
                        "confidence": 0.99853814,
                        "words": [
                            {
                                "word": "yep",
                                "start": 5.52,
                                "end": 6.02,
                                "confidence": 0.995842,
                                "punctuated_word": "Yep."
                            },
                            {
                                "word": "i",
                                "start": 7.04,
                                "end": 7.2799997,
                                "confidence": 0.51270777,
                                "punctuated_word": "I"
                            },
                            {
                                "word": "said",
                                "start": 7.2799997,
                                "end": 7.44,
                                "confidence": 0.96725076,
                                "punctuated_word": "said"
                            },
                            {
                                "word": "it",
                                "start": 7.44,
                                "end": 7.6,
                                "confidence": 0.9997285,
                                "punctuated_word": "it"
                            },
                            {
                                "word": "before",
                                "start": 7.6,
                                "end": 7.9199996,
                                "confidence": 0.7845937,
                                "punctuated_word": "before,"
                            },
                            {
                                "word": "and",
                                "start": 7.9199996,
                                "end": 8.16,
                                "confidence": 0.9998627,
                                "punctuated_word": "and"
                            },
                            {
                                "word": "i'll",
                                "start": 8.16,
                                "end": 8.32,
                                "confidence": 0.9998943,
                                "punctuated_word": "I'll"
                            },
                            {
                                "word": "say",
                                "start": 8.32,
                                "end": 8.48,
                                "confidence": 0.9996518,
                                "punctuated_word": "say"
                            },
                            {
                                "word": "it",
                                "start": 8.48,
                                "end": 8.639999,
                                "confidence": 0.99982834,
                                "punctuated_word": "it"
                            },
                            {
                                "word": "again",
                                "start": 8.639999,
                                "end": 9.139999,
                                "confidence": 0.97370684,
                                "punctuated_word": "again."
                            },
                            {
                                "word": "life",
                                "start": 9.991312,
                                "end": 10.391313,
                                "confidence": 0.99573696,
                                "punctuated_word": "Life"
                            },
                            {
                                "word": "moves",
                                "start": 10.391313,
                                "end": 10.711312,
                                "confidence": 0.9988587,
                                "punctuated_word": "moves"
                            },
                            {
                                "word": "pretty",
                                "start": 10.711312,
                                "end": 11.031313,
                                "confidence": 0.9996013,
                                "punctuated_word": "pretty"
                            },
                            {
                                "word": "fast",
                                "start": 11.031313,
                                "end": 11.531313,
                                "confidence": 0.9995537,
                                "punctuated_word": "fast."
                            },
                            {
                                "word": "you",
                                "start": 12.071312,
                                "end": 12.231313,
                                "confidence": 0.9514584,
                                "punctuated_word": "You"
                            },
                            {
                                "word": "don't",
                                "start": 12.231313,
                                "end": 12.4713125,
                                "confidence": 0.99988735,
                                "punctuated_word": "don't"
                            },
                            {
                                "word": "stop",
                                "start": 12.4713125,
                                "end": 12.711312,
                                "confidence": 0.99979633,
                                "punctuated_word": "stop"
                            },
                            {
                                "word": "and",
                                "start": 12.711312,
                                "end": 12.871312,
                                "confidence": 0.9987136,
                                "punctuated_word": "and"
                            },
                            {
                                "word": "look",
                                "start": 12.871312,
                                "end": 13.031313,
                                "confidence": 0.9996673,
                                "punctuated_word": "look"
                            },
                            {
                                "word": "around",
                                "start": 13.031313,
                                "end": 13.351313,
                                "confidence": 0.9995766,
                                "punctuated_word": "around"
                            },
                            {
                                "word": "once",
                                "start": 13.351313,
                                "end": 13.591312,
                                "confidence": 0.9981976,
                                "punctuated_word": "once"
                            },
                            {
                                "word": "in",
                                "start": 13.591312,
                                "end": 13.751312,
                                "confidence": 0.99853814,
                                "punctuated_word": "in"
                            },
                            {
                                "word": "a",
                                "start": 13.751312,
                                "end": 13.831312,
                                "confidence": 0.9861081,
                                "punctuated_word": "a"
                            },
                            {
                                "word": "while",
                                "start": 13.831312,
                                "end": 14.331312,
                                "confidence": 0.92629486,
                                "punctuated_word": "while,"
                            },
                            {
                                "word": "you",
                                "start": 14.631312,
                                "end": 14.791312,
                                "confidence": 0.99702364,
                                "punctuated_word": "you"
                            },
                            {
                                "word": "could",
                                "start": 14.791312,
                                "end": 14.951313,
                                "confidence": 0.9983536,
                                "punctuated_word": "could"
                            },
                            {
                                "word": "miss",
                                "start": 14.951313,
                                "end": 15.191313,
                                "confidence": 0.9984425,
                                "punctuated_word": "miss"
                            },
                            {
                                "word": "it",
                                "start": 15.191313,
                                "end": 15.691313,
                                "confidence": 0.9950415,
                                "punctuated_word": "it."
                            }
                        ],
                        "paragraphs": {
                            "transcript": "\nYep. I said it before, and I'll say it again. Life moves pretty fast. You don't stop and look around once in a while, you could miss it.",
                            "paragraphs": [
                                {
                                    "sentences": [
                                        {
                                            "text": "Yep.",
                                            "start": 5.52,
                                            "end": 6.02
                                        },
                                        {
                                            "text": "I said it before, and I'll say it again.",
                                            "start": 7.04,
                                            "end": 9.139999
                                        },
                                        {
                                            "text": "Life moves pretty fast.",
                                            "start": 9.991312,
                                            "end": 11.531313
                                        },
                                        {
                                            "text": "You don't stop and look around once in a while, you could miss it.",
                                            "start": 12.071312,
                                            "end": 15.691313
                                        }
                                    ],
                                    "start": 5.52,
                                    "end": 15.691313,
                                    "num_words": 28
                                }
                            ]
                        }
                    }
                ]
            }
        ]
    }
}
```

## Types of changes

What types of changes does your code introduce to the community Python SDK?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update or tests (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] I have lint'ed all of my code using repo standards
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

## Further comments

NA

## Summary by CodeRabbit

- **New Features**
	- Introduced new entities for handling WebSocket and REST responses, including `ListenWSWord`, `ListenWSAlternative`, `ListenWSChannel`, and their REST counterparts.
	- Enhanced data representation for WebSocket responses, improving the handling of words, alternatives, and channels.

- **Bug Fixes**
	- Clarified the handling of `_asyncio_thread` in the `Microphone` class for better state management.

- **Refactor**
	- Streamlined import statements across various modules to focus on new listening functionalities, removing outdated entities.

### CODE REVIEW

File: deepgram/clients/listen/v1/websocket/__init__.py [hunk 1]
```diff
@@ -21,11 +21,11 @@
...
-    Alternative,
     Hit,
     Search,
-    Channel,
-    Word,
     #### unique
     Metadata,
+    ListenWSWord,
+    ListenWSAlternative,
+    ListenWSChannel,
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply deepgram/clients/listen/v1/websocket/__init__.py:30]: This has been fixed in an updated commit.

File: deepgram/clients/listen/v1/websocket/response.py [hunk 2]
```diff
@@ -19,13 +19,78 @@
...
+    languages: Optional[List[str]] = field(
+        default=None, metadata=dataclass_config(exclude=lambda f: f is None)
+    )
+
+    def __getitem__(self, key):
+        _dict = self.to_dict()
+        if "words" in _dict:
+            _dict["words"] = [ListenWSWord.from_dict(words) for words in _dict["words"]]
+        return _dict[key]
+
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply deepgram/clients/listen/v1/websocket/response.py:69]: Ignoring this comment for now.

File: deepgram/clients/listen/v1/rest/response.py [hunk 3]
```diff
@@ -142,11 +139,24 @@ class SummaryV2(BaseResponse):
...
+    confidence: float = 0
+    punctuated_word: Optional[str] = field(
+        default=None, metadata=dataclass_config(exclude=lambda f: f is None)
+    )
+    speaker: Optional[int] = field(
+        default=None, metadata=dataclass_config(exclude=lambda f: f is None)
+    )
+    language: Optional[str] = field(
+        default=None, metadata=dataclass_config(exclude=lambda f: f is None)
+    )
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply deepgram/clients/listen/v1/rest/response.py:159]: This has been fixed in an updated commit.

File: deepgram/clients/listen/v1/websocket/response.py [hunk 4]
```diff
@@ -19,13 +19,78 @@
...
+        _dict = self.to_dict()
+        if "search" in _dict:
+            _dict["search"] = [Search.from_dict(search) for search in _dict["search"]]
+        if "alternatives" in _dict:
+            _dict["alternatives"] = [
+                ListenWSAlternative.from_dict(alternatives)
+                for alternatives in _dict["alternatives"]
+            ]
+        return _dict[key]
+
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply deepgram/clients/listen/v1/websocket/response.py:92]: Ignoring for now.

**Review by lukeocodes — APPROVED** [review]

**Pull request merged**